code/modules/clothing/suits/marine_armor.dm

```diff
@@ -282,6 +282,106 @@
+
+/datum/action/item_action/intel/toggle_motion_detector/action_activate()
+	. = ..()
+	var/obj/item/clothing/suit/storage/marine/medium/rto/intel/recon = holder_item
+	recon.toggle_motion_detector(usr)
+
+/datum/action/item_action/intel/toggle_motion_detector/proc/update_icon()
+	if(!holder_item)
+		return
+	var/obj/item/clothing/suit/storage/marine/medium/rto/intel/recon = holder_item
+	if(recon.motion_detector)
+		button.icon_state = "template_on"
+	else
+		button.icon_state = "template"
+
+/obj/item/clothing/suit/storage/marine/medium/rto/intel/process()
+	if(!motion_detector)
+		STOP_PROCESSING(SSobj, src)
+	if(motion_detector)
+		recycletime--
+		if(!recycletime)
+			recycletime = initial(recycletime)
+			proximity.refresh_blip_pool()
+		long_range_cooldown--
+		if(long_range_cooldown)
+			return
+		long_range_cooldown = initial(long_range_cooldown)
+		proximity.scan()
+
+/obj/item/clothing/suit/storage/marine/medium/rto/intel/proc/toggle_motion_detector(mob/user)
+	to_chat(user,SPAN_NOTICE("You [motion_detector? "<B>disable</b>" : "<B>enable</b>"] \the [src]'s motion detector."))
+	if(!motion_detector)
+		playsound(loc,'sound/items/detector_turn_on.ogg', 25, 1)
+	else
+		playsound(loc,'sound/items/detector_turn_off.ogg', 25, 1)
+	motion_detector = !motion_detector
+	var/datum/action/item_action/intel/toggle_motion_detector/TMD = locate(/datum/action/item_action/intel/toggle_motion_detector) in actions
+	TMD.update_icon()
+	motion_detector()
+
+/obj/item/clothing/suit/storage/marine/medium/rto/intel/proc/motion_detector()
+	if(motion_detector)
+		START_PROCESSING(SSobj, src)
+	if(!motion_detector)
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/clothing/suit/storage/marine/medium/rto/intel/mob_can_equip(mob/living/carbon/human/user, slot, disable_warning) //Thanks to Drathek for the help on this part!
+	if(!..())
+		return FALSE
+
+	// Only equip if uniform doesn't already have a utility accessory slot equipped
+	var/obj/item/clothing/under/uniform = user.w_uniform
+	var/accessory = locate(/obj/item/clothing/accessory/storage) in uniform.accessories
+	if(accessory)
+		to_chat(user, SPAN_WARNING("[src] can't be worn with [accessory]."))
+		return FALSE
+
+	return TRUE
+
+/obj/item/clothing/suit/storage/marine/medium/rto/intel/equipped(mob/user, slot, silent) //When XM4 is equipped this removes ACCESSORY_SLOT_UTILITY as a valid accessory for the uniform
+	. = ..()
+	var/mob/living/carbon/human/human = user
+	var/obj/item/clothing/under/uniform = human.w_uniform
+	if(slot == WEAR_JACKET)
+		uniform?.valid_accessory_slots -= ACCESSORY_SLOT_UTILITY
+
+/obj/item/clothing/suit/storage/marine/medium/rto/intel/unequipped(mob/user, slot) //When unequipped this adds the ACCESSORY_SLOT_UTILITY back as a valid accessory
+	. = ..()
+	var/mob/living/carbon/human/human = user
+	var/obj/item/clothing/under/uniform = human.w_uniform
+	uniform?.valid_accessory_slots += ACCESSORY_SLOT_UTILITY
```

Comment:
This should only be done if it originally had that slot; and only if the slot is WEAR_JACKET. See the patch I gave you previously.

Reply:
So the issue here is that in the off chance in an edge case it would add the slot to a uniform that didn't originally allow it?